### DISCUSSION TITLE
use mediainfo for video preview?

### DISCUSSION THREAD

**Discussion by ehula** [sxyazi/yazi#1264] [1. Q&A]
I would prefer to use `mediainfo` for preview of video files as opposed to the default `ffmpegthumbnailer`. Is there a way to do that?

**Comment by sxyazi** [comment]
AFAIK, `mediainfo` can only be used as a plain text previewer - it can't generate a thumbnail from a video. 

If it's what you want, you can easily achieve this by customizing the previewer. You can refer to the code in [`file.lua`](https://github.com/sxyazi/yazi/blob/main/yazi-plugin/preset/plugins/file.lua), just replace the `file` command with `mediainfo`.

See the previewer doc at: https://yazi-rs.github.io/docs/plugins/overview#previewer and https://yazi-rs.github.io/docs/configuration/yazi#plugin.previewers

---

Or maybe you want to combine both `mediainfo` and `ffmpegthumbnailer` to achieve mixing previewing just like what https://github.com/Sonico98/exifaudio.yazi does

**Reply by ehula** [reply]
Yes, I am wanting to replace the thumbnails of `ffmpegthumbnailer` with the text output of `mediainfo` for video file previews.

Combining `mediainfo` and `ffmpegthumbnailer` similar to `exifaudio.yazi` would be very interesting. Unfortunately, I don't have the chops to take this on.

**Comment by Ape** [comment]
Thanks for the great idea!

I made a plugin: https://github.com/Ape/mediainfo.yazi

**Comment by ehula** [comment]
This is awesome! Thanks for this. However, I can't seem to get it working. The thumbnails still show as before for images, audio and video, but no media metadata from mediainfo. I do have mediainfo installed and on the path.
